neutrinordp/xrdp-neutrinordp.c

```diff
@@ -1682,6 +1750,61 @@ lfreerdp_pre_connect(freerdp *instance)
         // | PERF_DISABLE_CURSOR_SHADOW | PERF_DISABLE_CURSORSETTINGS;
     }
 
+    /* Allow users or administrators to configure the mstsc experience settings. #1903 */
+
+    if ((mod->allow_client_experiencesettings == 1) &&
+            (mod->client_info.mcs_connection_type == CONNECTION_TYPE_AUTODETECT))
+    {
+        /* Do nothing to use the specified value. */
+    }
+    else if (mod->allow_client_experiencesettings == 1)
+    {
+        instance->settings->performance_flags =
+            (mod->client_info.rdp5_performanceflags &
+             /* Mask to avoid accepting invalid flags. */
+             (PERF_DISABLE_WALLPAPER |
+              PERF_DISABLE_FULLWINDOWDRAG |
+              PERF_DISABLE_MENUANIMATIONS |
+              PERF_DISABLE_THEMING |
+              PERF_DISABLE_CURSOR_SHADOW |
+              PERF_DISABLE_CURSORSETTINGS |
+              PERF_ENABLE_FONT_SMOOTHING |
+              PERF_ENABLE_DESKTOP_COMPOSITION));
+
+        LOG(LOG_LEVEL_DEBUG, "Use RDP client experience settings, "
+            "rdp5_performance_flags:[0x%08x], "
+            "masked performance_flags:[0x%08x]",
+            mod->client_info.rdp5_performanceflags,
+            instance->settings->performance_flags);
+
+        if (mod->client_info.rail_enable &&
+                (mod->client_info.rail_support_level > 0))
+        {
+            instance->settings->performance_flags |= (PERF_DISABLE_WALLPAPER |
+                    PERF_DISABLE_FULLWINDOWDRAG);
+            LOG(LOG_LEVEL_DEBUG, "Add in performance setting for Railsupport:"
+                "[0x%08x]", PERF_DISABLE_WALLPAPER |
+                PERF_DISABLE_FULLWINDOWDRAG);
+        }
+    }
+
+    LOG(LOG_LEVEL_DEBUG, "before overriding performance_flags:[0x%08x]",
+        instance->settings->performance_flags);
+    LOG(LOG_LEVEL_DEBUG, "perf_settings_override_mask:[0x%08x], "
+        "perf_settings_values_mask:[0x%08x]",
+        mod->perf_settings_override_mask,
+        mod->perf_settings_values_mask);
+
+    /* Clear bits for any overridden performance settings */
+    instance->settings->performance_flags =
```

Comment:
Better expressed as :-
```c
instance->settings->performance_flags &= ~mod->perf_settings_override_mask;
```

I think I suggested what you've got here - my apologies.

Reply:
I'm sorry.
It's my fault for pasting your mere example.
I will fix it.